sources/packages/backend/apps/workers/src/controllers/disbursement/_tests_/e2e/disbursement.controller.associateMSFAA.e2e-spec.ts

```diff
@@ -0,0 +1,659 @@
+      createFakeMSFAANumber(
+        {
+          student: student,
+        },
+        {
+          msfaaState: MSFAAStates.Signed,
+          msfaaInitialValues: {
+            offeringIntensity: OfferingIntensity.partTime,
+          },
+        },
+      ),
+    );
+
+    // Act
+    const associateMSFAAPayload = createFakeAssociateMSFAAPayload({
+      assessmentId: application.currentAssessment.id,
+    });
+    const saveResult = await disbursementController.associateMSFAA(
+      associateMSFAAPayload,
+    );
+
+    // Asserts
+    expect(FakeWorkerJobResult.getResultType(saveResult)).toBe(
+      MockedZeebeJobResult.Complete,
+    );
+
+    // Fetch MSFAA Number for the disbursement schedule assigned to the student in SIMS.
+    const disbursementSchedule = await db.disbursementSchedule.findOne({
+      select: {
+        id: true,
+        msfaaNumber: { id: true },
+      },
+      relations: {
+        msfaaNumber: true,
+        studentAssessment: true,
+      },
+      where: {
+        studentAssessment: { id: application.currentAssessment.id },
+      },
+    });
+    // Assert MSFAA Number.
+    expect(disbursementSchedule.msfaaNumber.id).toBe(savedMSFAANumber.id);
+  });
+
+  it("Should reuse the MSFAA Number from SIMS when MSFAA number is already generated and the request is pending for verification.", async () => {
+    // Arrange
+    // Create student and save fake application with disbursements.
+    const student = await saveFakeStudent(db.dataSource);
+    const currentMSFAA = createFakeMSFAANumber(
+      { student },
+      {
+        msfaaState: MSFAAStates.Pending,
+      },
+    );
+    await db.msfaaNumber.save(currentMSFAA);
+    const application = await saveFakeApplicationDisbursements(
+      db.dataSource,
+      { student },
+      {
+        applicationStatus: ApplicationStatus.Completed,
+      },
+    );
+
+    // Act
+    const associateMSFAAPayload = createFakeAssociateMSFAAPayload({
+      assessmentId: application.currentAssessment.id,
+    });
+    const saveResult = await disbursementController.associateMSFAA(
+      associateMSFAAPayload,
+    );
+
+    // Asserts
+    expect(FakeWorkerJobResult.getResultType(saveResult)).toBe(
+      MockedZeebeJobResult.Complete,
+    );
+
+    // Fetch MSFAA Number for the disbursement schedule assigned to the student in SIMS.
+    const disbursementSchedule = await db.disbursementSchedule.findOne({
+      select: {
+        id: true,
+        msfaaNumber: { id: true },
+      },
+      relations: {
+        msfaaNumber: true,
+        studentAssessment: true,
+      },
+      where: {
+        studentAssessment: { id: application.currentAssessment.id },
+      },
+    });
+    // Assert MSFAA Number.
+    expect(disbursementSchedule.msfaaNumber.id).toBe(currentMSFAA.id);
+  });
+
+  it("Should reuse the part-time MSFAA number from SFAS part-time applications creating and activating it in SIMS when a valid MSFAA was found in SFAS.", async () => {
```

Comment:
A pending MSFAA can be created for the same student and same intensity to ensure the SFAS one will take precedence.

An assertion can also be added to ensure the pending MSFAA was canceled.